src/main/java/us/kbase/auth2/lib/UserName.java

```diff
@@ -35,8 +39,8 @@ public class UserName extends Name {
 		}
 	}
 	
-	private static final String INVALID_CHARS_REGEX = "[^a-z\\d_]+";
-	private final static Pattern INVALID_CHARS = Pattern.compile(INVALID_CHARS_REGEX);
+	private static final Pattern FORCE_ALPHA_FIRST_CHAR = Pattern.compile("^[^a-z]+");
+	private final static Pattern INVALID_CHARS = Pattern.compile("[^a-z\\d_]+");
```

Comment:
can you use `\\W+`, since `\w` is equivalent to `[a-z0-9_]` in most regex implementations?

Reply:
That would work, but my personal preference is to use the more explicit form where you don't have to look up what `W` means